src/aspects/ec2-automated-shutdown/index.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { join } from 'path';
+import { Stack, Duration, IAspect, Aws } from 'aws-cdk-lib';
+import { Metric, Alarm, ComparisonOperator } from 'aws-cdk-lib/aws-cloudwatch';
+import { LambdaAction } from 'aws-cdk-lib/aws-cloudwatch-actions';
+import { Instance, CfnInstance } from 'aws-cdk-lib/aws-ec2';
+import { PolicyStatement, Effect } from 'aws-cdk-lib/aws-iam';
+import { IKey } from 'aws-cdk-lib/aws-kms';
+import { Code, Runtime } from 'aws-cdk-lib/aws-lambda';
+import { IConstruct } from 'constructs';
+import { SecureFunction } from '../../constructs/secure-function';
+import { LambdaConfiguration } from '../../types/lambda-configuration';
+
+export interface CustomMetricConfig {
+    readonly MetricName: string;
```

Comment:
all these interface props must be camelCase. Also, can we get docstrings on this CustomMetricConfig interface?